packages/guides/src/Intersphinx/JsonLoader.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\Intersphinx;
+
+use JsonException;
+use RuntimeException;
+
+use function file_get_contents;
+use function is_array;
+use function json_decode;
+
+use const JSON_THROW_ON_ERROR;
+
+class JsonLoader
+{
+    /** @return array<mixed> */
+    public function loadJsonFromUrl(string $url): array
+    {
+        $jsonString = file_get_contents($url);
```

Comment:
Please keep in mind that `file_get_contents` will only accept URLs when ` allow_url_fopen` in the PHP.ini is enabled. This is the default, but we might want to add an alternative way to do this.

Would the usage of an HTTP client interface be an option? https://www.php-fig.org/psr/psr-18/

We could add a basic implementation of this interface, just doing a `file_get_contents` as a sane default. By using HTTP clients we could even allow people to link to documentationsets with basic authentication, or more advanced authentication if they implement their own way. 

All these are just suggestions, no need to include that right now.